<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Annotate Peaks with DNA Interaction Networks Using Graph Clustering — annoLinker • annoLinker</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Annotate Peaks with DNA Interaction Networks Using Graph Clustering — annoLinker"><meta property="og:description" content="Fast annotation of genomic peaks using DNA interaction data by building
interaction networks with igraph. Peaks overlapping any node in a connected
subgraph are annotated with all genes in that subgraph."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">annoLinker</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/annoLinker.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jianhong/annoLinker" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Annotate Peaks with DNA Interaction Networks Using Graph Clustering</h1>
    <small class="dont-index">Source: <a href="https://github.com/jianhong/annoLinker/blob/devel/R/annoLinker.R" class="external-link"><code>R/annoLinker.R</code></a></small>
    <div class="hidden name"><code>annoLinker.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Fast annotation of genomic peaks using DNA interaction data by building
interaction networks with igraph. Peaks overlapping any node in a connected
subgraph are annotated with all genes in that subgraph.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">annoLinker</span><span class="op">(</span></span>
<span>  <span class="va">peaks</span>,</span>
<span>  <span class="va">annoData</span>,</span>
<span>  <span class="va">interactions</span>,</span>
<span>  bindingType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"startSite"</span>, <span class="st">"body"</span>, <span class="st">"endSite"</span><span class="op">)</span>,</span>
<span>  bindingRegion <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5000</span>, <span class="fl">5000</span><span class="op">)</span>,</span>
<span>  cluster_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"components"</span>, <span class="st">"louvain"</span>, <span class="st">"walktrap"</span>, <span class="st">"infomap"</span><span class="op">)</span>,</span>
<span>  extend_anchors <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  interactionDistanceRange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10000</span>, <span class="fl">1e+07</span><span class="op">)</span>,</span>
<span>  addEvidence <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-peaks">peaks<a class="anchor" aria-label="anchor" href="#arg-peaks"></a></dt>
<dd><p>GRanges object containing peak regions</p></dd>


<dt id="arg-annodata">annoData<a class="anchor" aria-label="anchor" href="#arg-annodata"></a></dt>
<dd><p>annoGR or GRanges object with gene annotations</p></dd>


<dt id="arg-interactions">interactions<a class="anchor" aria-label="anchor" href="#arg-interactions"></a></dt>
<dd><p>GInteractions, or Pairs object with
interaction data (e.g., Hi-C, ChIA-PET)</p></dd>


<dt id="arg-bindingtype">bindingType<a class="anchor" aria-label="anchor" href="#arg-bindingtype"></a></dt>
<dd><p>Character, one of "startSite", "body", or "endSite"</p></dd>


<dt id="arg-bindingregion">bindingRegion<a class="anchor" aria-label="anchor" href="#arg-bindingregion"></a></dt>
<dd><p>Numeric vector of length 2 defining promoter window
(e.g., c(-5000, 5000))</p></dd>


<dt id="arg-cluster-method">cluster_method<a class="anchor" aria-label="anchor" href="#arg-cluster-method"></a></dt>
<dd><p>Character, clustering method: "components" (connected
components), "louvain", "walktrap", or "infomap"</p></dd>


<dt id="arg-extend-anchors">extend_anchors<a class="anchor" aria-label="anchor" href="#arg-extend-anchors"></a></dt>
<dd><p>Integer, bp to extend interaction anchors for
overlap detection (default: 0)</p></dd>


<dt id="arg-interactiondistancerange">interactionDistanceRange<a class="anchor" aria-label="anchor" href="#arg-interactiondistancerange"></a></dt>
<dd><p>Numeric vector of length 2 defining the minimal
and maximal distance of interactions. This is used to make sure the annotations
are not supper far away.</p></dd>


<dt id="arg-addevidence">addEvidence<a class="anchor" aria-label="anchor" href="#arg-addevidence"></a></dt>
<dd><p>Logical, add evidence to the metadata or not.</p></dd>


<dt id="arg-parallel">parallel<a class="anchor" aria-label="anchor" href="#arg-parallel"></a></dt>
<dd><p>Logical, use future_lapply to do parallel computing or not.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Logical, print the message or not</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Parameters for cluster. see <a href="https://r.igraph.org/reference/cluster_louvain.html" class="external-link">cluster_louvain</a>,
<a href="https://r.igraph.org/reference/cluster_walktrap.html" class="external-link">cluster_walktrap</a>, and <a href="https://r.igraph.org/reference/cluster_infomap.html" class="external-link">cluster_infomap</a>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>An <a href="annoLinkerResult-class.html">annoLinkerResult</a> object
or NULL if no annotations found</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## read the peaks and interactions</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: generics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'generics'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     setequal, union</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'BiocGenerics'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:annoLinker':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:stats':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, is.unsorted, lapply,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     mapply, match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rank, rbind, rownames, sapply, saveRDS, table, tapply, unique,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'S4Vectors'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:utils':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     findMatches</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Seqinfo</span>
<span class="r-in"><span><span class="va">extPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, package<span class="op">=</span><span class="st">'annoLinker'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">extPath</span>, <span class="st">'peaks.bed'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">interactions</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">extPath</span>, <span class="st">'interaction.bedpe'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Drerio.UCSC.danRer10.refGene</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicFeatures</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationDbi</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-in"><span><span class="va">annoData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">TxDb.Drerio.UCSC.danRer10.refGene</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   32 genes were dropped because they have exons located on both strands of the</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   same reference sequence or on more than one reference sequence, so cannot be</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   represented by a single genomic range.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Use 'single.strand.genes.only=FALSE' to get all the genes in a GRangesList</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   object, or use suppressMessages() to suppress this message.</span>
<span class="r-in"><span><span class="va">anno</span> <span class="op">&lt;-</span> <span class="fu">annoLinker</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">annoData</span>, <span class="va">interactions</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Step 1/4: Building interaction network graph...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Graph has 3753 nodes and 2500 edges in 1264 clusters, on average 2.969146 nodes in each cluster</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Step 2/4: Finding interaction anchors overlapping genes and peaks...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Step 3/4: Finding genes and peaks in the same cluster...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Step 4/4: Annotating peaks with gene clusters...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Annotated 2865 peaks with gene clusters</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jianhong Ou, Kenneth Poss.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

