<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>annoLinker Vignette: Annotating genomic regions through chromatin interaction links â€¢ annoLinker</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="annoLinker Vignette: Annotating genomic regions through chromatin interaction links">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">annoLinker</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/annoLinker.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jianhong/annoLinker" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><link href="annoLinker_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="annoLinker_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="annoLinker_files/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="annoLinker_files/vis-9.1.0/vis-network.min.js"></script><script src="annoLinker_files/visNetwork-binding-2.1.4/visNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>annoLinker Vignette: Annotating genomic regions
through chromatin interaction links</h1>
                        <h4 data-toc-skip class="author">Jianhong
Ou</h4>
            
            <h4 data-toc-skip class="date">30 October 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jianhong/annoLinker/blob/devel/vignettes/annoLinker.Rmd" class="external-link"><code>vignettes/annoLinker.Rmd</code></a></small>
      <div class="hidden name"><code>annoLinker.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Annotating genomic regions through chromatin interaction
      links.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Functional annotation of genomic regions is a key step in
understanding regulatory mechanisms underlying gene expression and
phenotype variation. Traditional annotation methods typically rely on
the linear proximity of genomic features, such as linking peaks to the
nearest gene, which can overlook the complex three-dimensional
organization of the genome. Recent advances in chromatin conformation
capture technologies (e.g., Hi-C, ChIA-PET, PLAC-seq, HiCAR) have
revealed that regulatory elements often interact with distant genomic
loci through chromatin loops, bringing enhancers, promoters, and other
elements into physical contact.</p>
<p><code>annoLinker</code> is designed to bridge this gap by annotating
genomic regions through chromatin interaction links. Rather than relying
solely on linear genomic distance, <code>annoLinker</code> integrates
chromatin contact information to establish biologically meaningful
connections between regulatory regions and their potential target genes.
This approach complements traditional annotation methods by enabling the
functional assignment of distal elements such as enhancers and
silencers.</p>
<p>The package provides a streamlined workflow to link user-defined
genomic regions with genomic interactions (loops), annotate the
associated genes, and visualize the resulting interaction networks. By
leveraging interaction data, <code>annoLinker</code> enables users to
explore functional relationships that are invisible in linear genome
coordinates and to generate more comprehensive biological
interpretations from genomic assays such as ATAC-seq, ChIP-seq, or
Methyl-seq.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"jianhong/annoLinker"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<p>The input to <code>annoLinker</code> consists of three components:
the genomic regions of interest to be annotated, the annotation data,
and the chromatin interaction information. Both the peak list and
annotation data should be provided as <code>GRanges</code> objects,
while genomic interactions can be supplied as either
<code>GInteractions</code> or <code>Pairs</code> objects.</p>
<p>By default, <code>annoLinker</code> annotates peaks that fall within
the <em>promoter region</em>, defined as 5 kb upstream to 5 kb
downstream of the annotated features. Users can optionally modify this
definition to annotate regions within the <em>gene body</em> or
<em>downstream</em> segments instead.</p>
<p>The output is returned as an <code>annoLinkerResult</code> object,
which can be easily converted into a <code>GRanges</code> or
<code>data.frame</code> for downstream analysis and visualization.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jianhong/annoLinker" class="external-link">annoLinker</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Drerio.UCSC.danRer10.refGene</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Dr.eg.db</span><span class="op">)</span></span>
<span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">TxDb.Drerio.UCSC.danRer10.refGene</span></span>
<span><span class="va">org</span> <span class="op">&lt;-</span> <span class="va">org.Dr.eg.db</span></span>
<span><span class="va">extPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, package<span class="op">=</span><span class="st">'annoLinker'</span><span class="op">)</span></span>
<span><span class="co">## load peaks</span></span>
<span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">extPath</span>, <span class="st">'peaks.bed'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## load interactions</span></span>
<span><span class="va">interactions</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">extPath</span>, <span class="st">'interaction.bedpe'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## load annotation data</span></span>
<span><span class="va">annoData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span></span>
<span><span class="va">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annoLinker.html">annoLinker</a></span><span class="op">(</span><span class="va">peaks</span>, <span class="va">annoData</span>, <span class="va">interactions</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "annoLinkerResult"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "annoLinker"</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">anno</span>, n<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 2 ranges and 7 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand | feature_name feature_start</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;character&gt;     &lt;integer&gt;</span></span>
<span><span class="co">##   [1]     chr1 45062825-45063844      * |    100151507      44660095</span></span>
<span><span class="co">##   [2]     chr1 45062825-45063844      * |    100038798      45052323</span></span>
<span><span class="co">##       feature_end feature_strand     gene_id               peak_bin</span></span>
<span><span class="co">##         &lt;integer&gt;          &lt;Rle&gt; &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">##   [1]    44674627              +   100151507 chr1:45060001-45070000</span></span>
<span><span class="co">##   [2]    45060215              -   100038798 chr1:45060001-45070000</span></span>
<span><span class="co">##                  feature_bin</span></span>
<span><span class="co">##                  &lt;character&gt;</span></span>
<span><span class="co">##   [1] chr1:44660001-44670000</span></span>
<span><span class="co">##   [2] chr1:45050001-45060000</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/annoLinkerResult-class.html">anno_peaks</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 2 ranges and 7 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand | feature_name feature_start</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;character&gt;     &lt;integer&gt;</span></span>
<span><span class="co">##   [1]     chr1 45062825-45063844      * |    100151507      44660095</span></span>
<span><span class="co">##   [2]     chr1 45062825-45063844      * |    100038798      45052323</span></span>
<span><span class="co">##       feature_end feature_strand     gene_id               peak_bin</span></span>
<span><span class="co">##         &lt;integer&gt;          &lt;Rle&gt; &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">##   [1]    44674627              +   100151507 chr1:45060001-45070000</span></span>
<span><span class="co">##   [2]    45060215              -   100038798 chr1:45060001-45070000</span></span>
<span><span class="co">##                  feature_bin</span></span>
<span><span class="co">##                  &lt;character&gt;</span></span>
<span><span class="co">##   [1] chr1:44660001-44670000</span></span>
<span><span class="co">##   [2] chr1:45050001-45060000</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">anno</span>, <span class="st">'GRanges'</span><span class="op">)</span>, n<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 2 ranges and 7 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand | feature_name feature_start</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;character&gt;     &lt;integer&gt;</span></span>
<span><span class="co">##   [1]     chr1 45062825-45063844      * |    100151507      44660095</span></span>
<span><span class="co">##   [2]     chr1 45062825-45063844      * |    100038798      45052323</span></span>
<span><span class="co">##       feature_end feature_strand     gene_id               peak_bin</span></span>
<span><span class="co">##         &lt;integer&gt;          &lt;Rle&gt; &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">##   [1]    44674627              +   100151507 chr1:45060001-45070000</span></span>
<span><span class="co">##   [2]    45060215              -   100038798 chr1:45060001-45070000</span></span>
<span><span class="co">##                  feature_bin</span></span>
<span><span class="co">##                  &lt;character&gt;</span></span>
<span><span class="co">##   [1] chr1:44660001-44670000</span></span>
<span><span class="co">##   [2] chr1:45050001-45060000</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>The <code><a href="../reference/plotEvidence.html">plotEvidence()</a></code> function allows users to visualize
the annotation evidence chain, illustrating how genomic regions are
linked through chromatin interactions. Users can plot the annotation
evidence as either a <em>network graph</em> or a <em>genomic track</em>
integrated with gene annotation information.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## plot the evidence for the first annotation</span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="fu"><a href="../reference/plotEvidence.html">plotEvidence</a></span><span class="op">(</span><span class="va">anno</span>, event<span class="op">=</span><span class="fl">1</span>,</span>
<span>             output<span class="op">=</span><span class="st">'htmlWidget'</span><span class="op">)</span></span></code></pre></div>
<div class="visNetwork html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"nodes":{"id":["chr1:45130001-45140000","chr1:45050001-45060000","chr1:44900001-44910000","chr1:45060001-45070000","chr1:45040001-45050000","chr1:44550001-44560000","chr1:45160001-45170000","chr1:44280001-44290000","chr1:43450001-43460000","chr1:44660001-44670000","chr1:46800001-46810000","chr1:46270001-46280000","chr1:45290001-45300000","chr1:45062825-45063844","chr1:44660095-44674627:+"],"label":["chr1:45130001-45140000","chr1:45050001-45060000","chr1:44900001-44910000","chr1:45060001-45070000","chr1:45040001-45050000","chr1:44550001-44560000","chr1:45160001-45170000","chr1:44280001-44290000","chr1:43450001-43460000","chr1:44660001-44670000","chr1:46800001-46810000","chr1:46270001-46280000","chr1:45290001-45300000","peak","feature"],"title":["chr1:45130001-45140000","chr1:45050001-45060000","chr1:44900001-44910000","chr1:45060001-45070000","chr1:45040001-45050000","chr1:44550001-44560000","chr1:45160001-45170000","chr1:44280001-44290000","chr1:43450001-43460000","chr1:44660001-44670000","chr1:46800001-46810000","chr1:46270001-46280000","chr1:45290001-45300000","chr1:45062825-45063844","chr1:44660095-44674627:+"],"color":["lightgray","lightgray","lightgray","tomato","tomato","lightgray","lightgray","lightgray","lightgray","tomato","lightgray","tomato","lightgray","darkgreen","brown"],"size":[15,15,15,20,20,15,15,15,15,20,15,20,15,25,25]},"edges":{"from":["chr1:45130001-45140000","chr1:45130001-45140000","chr1:44900001-44910000","chr1:44900001-44910000","chr1:44900001-44910000","chr1:45040001-45050000","chr1:45130001-45140000","chr1:45160001-45170000","chr1:45130001-45140000","chr1:45060001-45070000","chr1:45040001-45050000","chr1:45040001-45050000","chr1:45060001-45070000","chr1:44660001-44670000"],"to":["chr1:45050001-45060000","chr1:44550001-44560000","chr1:45160001-45170000","chr1:44280001-44290000","chr1:43450001-43460000","chr1:44660001-44670000","chr1:46800001-46810000","chr1:46800001-46810000","chr1:46270001-46280000","chr1:46270001-46280000","chr1:46270001-46280000","chr1:45290001-45300000","chr1:45062825-45063844","chr1:44660095-44674627:+"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false}},"groups":null,"width":null,"height":null,"idselection":{"enabled":true,"style":"width: 150px; height: 26px","useLabels":true,"main":"Select by id"},"byselection":{"enabled":false,"style":"width: 150px; height: 26px","multiple":false,"hideColor":"rgba(200,200,200,0.5)","highlight":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","highlight":{"enabled":true,"hoverNearest":false,"degree":1,"algorithm":"all","hideColor":"rgba(200,200,200,0.5)","labelOnly":true},"collapse":{"enabled":false,"fit":false,"resetHighlight":true,"clusterOptions":null,"keepCoord":true,"labelSuffix":"(cluster)"}},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotEvidence.html">plotEvidence</a></span><span class="op">(</span><span class="va">anno</span>, event<span class="op">=</span><span class="fl">1</span>,</span>
<span>             output<span class="op">=</span><span class="st">'trackPlot'</span>,</span>
<span>             txdb<span class="op">=</span><span class="va">TxDb.Drerio.UCSC.danRer10.refGene</span>,</span>
<span>             org<span class="op">=</span><span class="va">org.Dr.eg.db</span><span class="op">)</span></span></code></pre></div>
<p><img src="annoLinker_files/figure-html/datavisualization-2.png" width="700"></p>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<pre><code><span><span class="co">## R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] org.Dr.eg.db_3.22.0                    </span></span>
<span><span class="co">##  [2] TxDb.Drerio.UCSC.danRer10.refGene_3.4.6</span></span>
<span><span class="co">##  [3] GenomicFeatures_1.61.8                 </span></span>
<span><span class="co">##  [4] AnnotationDbi_1.71.2                   </span></span>
<span><span class="co">##  [5] Biobase_2.69.1                         </span></span>
<span><span class="co">##  [6] rtracklayer_1.69.1                     </span></span>
<span><span class="co">##  [7] GenomicRanges_1.61.8                   </span></span>
<span><span class="co">##  [8] Seqinfo_0.99.4                         </span></span>
<span><span class="co">##  [9] IRanges_2.43.8                         </span></span>
<span><span class="co">## [10] S4Vectors_0.47.6                       </span></span>
<span><span class="co">## [11] BiocGenerics_0.55.4                    </span></span>
<span><span class="co">## [12] generics_0.1.4                         </span></span>
<span><span class="co">## [13] annoLinker_0.99.1                      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] strawr_0.0.92               RColorBrewer_1.1-3         </span></span>
<span><span class="co">##   [3] rstudioapi_0.17.1           jsonlite_2.0.0             </span></span>
<span><span class="co">##   [5] magrittr_2.0.4              farver_2.1.2               </span></span>
<span><span class="co">##   [7] rmarkdown_2.30              fs_1.6.6                   </span></span>
<span><span class="co">##   [9] BiocIO_1.19.0               ragg_1.5.0                 </span></span>
<span><span class="co">##  [11] vctrs_0.6.5                 memoise_2.0.1              </span></span>
<span><span class="co">##  [13] Rsamtools_2.25.3            RCurl_1.98-1.17            </span></span>
<span><span class="co">##  [15] base64enc_0.1-3             htmltools_0.5.8.1          </span></span>
<span><span class="co">##  [17] S4Arrays_1.9.2              progress_1.2.3             </span></span>
<span><span class="co">##  [19] curl_7.0.0                  Rhdf5lib_1.31.1            </span></span>
<span><span class="co">##  [21] rhdf5_2.53.6                SparseArray_1.9.1          </span></span>
<span><span class="co">##  [23] Formula_1.2-5               sass_0.4.10                </span></span>
<span><span class="co">##  [25] parallelly_1.45.1           bslib_0.9.0                </span></span>
<span><span class="co">##  [27] htmlwidgets_1.6.4           desc_1.4.3                 </span></span>
<span><span class="co">##  [29] Gviz_1.53.1                 httr2_1.2.1                </span></span>
<span><span class="co">##  [31] cachem_1.1.0                GenomicAlignments_1.45.5   </span></span>
<span><span class="co">##  [33] igraph_2.2.1                lifecycle_1.0.4            </span></span>
<span><span class="co">##  [35] pkgconfig_2.0.3             Matrix_1.7-4               </span></span>
<span><span class="co">##  [37] R6_2.6.1                    fastmap_1.2.0              </span></span>
<span><span class="co">##  [39] MatrixGenerics_1.21.0       future_1.67.0              </span></span>
<span><span class="co">##  [41] digest_0.6.37               colorspace_2.1-2           </span></span>
<span><span class="co">##  [43] textshaping_1.0.4           Hmisc_5.2-4                </span></span>
<span><span class="co">##  [45] RSQLite_2.4.3               filelock_1.0.3             </span></span>
<span><span class="co">##  [47] progressr_0.17.0            httr_1.4.7                 </span></span>
<span><span class="co">##  [49] abind_1.4-8                 compiler_4.5.1             </span></span>
<span><span class="co">##  [51] bit64_4.6.0-1               htmlTable_2.4.3            </span></span>
<span><span class="co">##  [53] S7_0.2.0                    backports_1.5.0            </span></span>
<span><span class="co">##  [55] BiocParallel_1.43.4         DBI_1.2.3                  </span></span>
<span><span class="co">##  [57] biomaRt_2.65.16             rappdirs_0.3.3             </span></span>
<span><span class="co">##  [59] DelayedArray_0.35.4         rjson_0.2.23               </span></span>
<span><span class="co">##  [61] tools_4.5.1                 foreign_0.8-90             </span></span>
<span><span class="co">##  [63] future.apply_1.20.0         nnet_7.3-20                </span></span>
<span><span class="co">##  [65] glue_1.8.0                  restfulr_0.0.16            </span></span>
<span><span class="co">##  [67] InteractionSet_1.37.1       rhdf5filters_1.21.4        </span></span>
<span><span class="co">##  [69] grid_4.5.1                  checkmate_2.3.3            </span></span>
<span><span class="co">##  [71] cluster_2.1.8.1             gtable_0.3.6               </span></span>
<span><span class="co">##  [73] BSgenome_1.77.3             trackViewer_1.45.3         </span></span>
<span><span class="co">##  [75] ensembldb_2.33.2            data.table_1.17.8          </span></span>
<span><span class="co">##  [77] hms_1.1.4                   XVector_0.49.3             </span></span>
<span><span class="co">##  [79] pillar_1.11.1               stringr_1.5.2              </span></span>
<span><span class="co">##  [81] dplyr_1.1.4                 BiocFileCache_2.99.6       </span></span>
<span><span class="co">##  [83] lattice_0.22-7              deldir_2.0-4               </span></span>
<span><span class="co">##  [85] bit_4.6.0                   biovizBase_1.57.1          </span></span>
<span><span class="co">##  [87] tidyselect_1.2.1            Biostrings_2.77.2          </span></span>
<span><span class="co">##  [89] knitr_1.50                  gridExtra_2.3              </span></span>
<span><span class="co">##  [91] ProtGenerics_1.41.0         SummarizedExperiment_1.39.2</span></span>
<span><span class="co">##  [93] xfun_0.53                   matrixStats_1.5.0          </span></span>
<span><span class="co">##  [95] visNetwork_2.1.4            stringi_1.8.7              </span></span>
<span><span class="co">##  [97] UCSC.utils_1.5.1            lazyeval_0.2.2             </span></span>
<span><span class="co">##  [99] yaml_2.3.10                 evaluate_1.0.5             </span></span>
<span><span class="co">## [101] codetools_0.2-20            cigarillo_0.99.2           </span></span>
<span><span class="co">## [103] interp_1.1-6                tibble_3.3.0               </span></span>
<span><span class="co">## [105] BiocManager_1.30.26         cli_3.6.5                  </span></span>
<span><span class="co">## [107] rpart_4.1.24                systemfonts_1.3.1          </span></span>
<span><span class="co">## [109] jquerylib_0.1.4             dichromat_2.0-0.1          </span></span>
<span><span class="co">## [111] Rcpp_1.1.0                  GenomeInfoDb_1.45.13       </span></span>
<span><span class="co">## [113] globals_0.18.0              grImport_0.9-7             </span></span>
<span><span class="co">## [115] dbplyr_2.5.1                png_0.1-8                  </span></span>
<span><span class="co">## [117] XML_3.99-0.19               parallel_4.5.1             </span></span>
<span><span class="co">## [119] pkgdown_2.1.3               ggplot2_4.0.0              </span></span>
<span><span class="co">## [121] blob_1.2.4                  prettyunits_1.2.0          </span></span>
<span><span class="co">## [123] jpeg_0.1-11                 latticeExtra_0.6-31        </span></span>
<span><span class="co">## [125] AnnotationFilter_1.33.0     bitops_1.0-9               </span></span>
<span><span class="co">## [127] txdbmaker_1.5.7             listenv_0.9.1              </span></span>
<span><span class="co">## [129] VariantAnnotation_1.55.2    scales_1.4.0               </span></span>
<span><span class="co">## [131] crayon_1.5.3                BiocStyle_2.37.1           </span></span>
<span><span class="co">## [133] rlang_1.1.6                 KEGGREST_1.49.2</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jianhong Ou, Kenneth Poss.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
